apps/explorer/lib/explorer/chain.ex

```diff
@@ -7029,18 +7029,15 @@ defmodule Explorer.Chain do
 
   def get_implementation_abi_from_proxy(proxy_address_hash, abi)
       when not is_nil(proxy_address_hash) and not is_nil(abi) do
-    implementation_method_abi =
-      abi
-      |> Enum.find(fn method ->
-        Map.get(method, "name") == "implementation" ||
-          master_copy_pattern?(method)
-      end)
-
-    if implementation_method_abi do
-      implementation_address_hash_string = get_implementation_address_hash(proxy_address_hash, abi)
+    if length(abi) > 0 do
```

Comment:
Which case does this check handle actually?

In any case we could use matching on `[]` like:
```
def get_implementation_abi_from_proxy(proxy_address_hash, []) when not is_nil(proxy_address_hash) do
  []
end
```
similar to the clause below.

Reply:
So we are checking that there is not nil, but when we have [] `is_nil` will be true so we need to check the length of the List to verify that we don't have an empty abi.

Reply:
Yeah, I'm just curious what has changed since we didn't change the way this function is called, so if that wasn't a problem before, why did we need to add the check.

Reply:
Because the `proxy_contract?` is different and when you call it is returning a false when is not instead of returning the []

Reply:
Then it looks like we need to add another clause for `proxy_contract?` like:
```
def proxy_contract?(_address_hash, []), do: false
```
instead of this check here. @rkachowski what do you think?

Reply:
yeah that would work, we could also do 

```elixir
def proxy_contract?(_address_hash, abi) when abi in [nil, false, []], do: false
```

if we are paranoid